### ISSUE TITLE
Use GraphQL for CRUD metafields operations

### ISSUE THREAD

**Issue by husseyexplores** [husseyexplores/poshify-utilities#1]
**Tldr;** Getting csrf token from the DOM is not reliable and can be changed anytime which will break the app. Use GraphQL instead, which doesn't require csrf token.

---

Currently, metafields CRUD operations use the REST API. And CUD operations require csrf token (`x-csrf-token` value in the headers) to complete which is not always present in the document. 

If it exists in the DOM, no problem, we use that token. But if the token does not exist (due to recent Shopify admin/auth changes) we first fetch `/admin/articles` page which contains csrf token in the DOM and use it from there.

Since we don't know for how long we can rely upon this and when `/admin/article` gets updated like other admin routes and stops exposing csrf token, we need to use GraphQL endpoint which I last checked, doesn't require csrf token.

But, the GraphQL API is not in parity with the REST API yet, so as of now, GraphQL doesn't support mutations on Shop, Article, Page, Blog resources. So we cannot switch to GraphQL yet.

This issue will be addressed when the Shopify GraphQL API starts supporting the above resources for mutations.
